bpf/bpf_host.c

```diff
@@ -1651,7 +1651,23 @@ int cil_to_host(struct __ctx_buff *ctx)
 	 * to mark as PACKET_OTHERHOST and drop.
 	 */
 	ctx_change_type(ctx, PACKET_HOST);
-#endif
+
+# ifdef ENABLE_NODEPORT
+	if (!validate_ethertype(ctx, &proto)) {
+		ret = DROP_UNSUPPORTED_L2;
+		goto out;
+	}
+	/* handle_nat_fwd() tail calls in the majority of cases, so control
```

Comment:
I'm pretty sure that the inline/tail-call decision is currently based on expected program size in `to-netdev`. Maybe we can optimize this a bit further in the future, and then inline the logic here.

Reply:
Interesting, yes.